test/jdk/java/util/Currency/CheckDataVersion.java

```diff
@@ -63,33 +61,23 @@ static void check() {
                         break;
                     }
                 }
+                InputStream JREdata = Currency.class.getModule().getResourceAsStream("java/util/currency.data");
+                DataInputStream dis = new DataInputStream(JREdata);
+                int magic = dis.readInt();
+                if (magic != 0x43757244) {
+                    throw new RuntimeException("The magic number in the JRE's currency data is incorrect.  Expected: 0x43757244, Got: 0x"+magic);
+                }
+                int fileVerNumber = dis.readInt();
+                int dataVerNumber = dis.readInt();
+                if (Integer.parseInt(fileVersion) != fileVerNumber ||
+                        Integer.parseInt(dataVersion) != dataVerNumber) {
+                    throw new RuntimeException("Test data and JRE's currency data are inconsistent.  test: (file: "+fileVersion+" data: "+dataVersion+"), JRE: (file: "+fileVerNumber+" data: "+dataVerNumber+")");
+                }
+                System.out.println("test: (file: "+fileVersion+" data: "+dataVersion+"), JRE: (file: "+fileVerNumber+" data: "+dataVerNumber+")");
             } catch (IOException ioe) {
-                throw new RuntimeException(ioe);
+                throw new RuntimeException(
+                        "CheckDataVersion was not set up properly: " + ioe);
```

Comment:
IOException could happen not by the test setup failure, but for the case the `currency.data` file in the JDK is missing or corrupt, so if you want to give extra message to the exception, needs some rewording. Also, I prefer `ioe` should be the `cause` and not to be consumed in the message.

Reply:
Ah, I meant that if _currency.data_ was missing/corrupted, it was a set up failure because we can not test the contents properly. Although, I suppose that just means the test fails, not fails to setup properly. Updated the message to be more clear, and also used the 2 arg ctor to retain the IOE as you suggested.